native/spark-expr/Cargo.toml

```diff
@@ -0,0 +1,38 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+[package]
+name = "datafusion-comet-expr"
```

Comment:
I wasn't sure whether to name this `datafusion-spark-expr` or `datafusion-comet-expr`. I think either could work.

I went with the latter because the versioning of this crate will be different from the version numbers used for other `datafusion-*` crates from the core project and thought that may cause confusion.

I also think it is unlikely that we would move this crate to the core project because we would want the implementation and tests to live in the same repo and the tests depend on Apache Spark, which would be a huge burrden to add to the core project.

cc @alamb in case you have any opinions on this.

Reply:
I have been thinking more about this, and I also now think we should name it `datafusion-spark-expr` because the functions are not specific to the Comet project. I will make that change.

I don't think we would ever want to move the code into `datafusion-contrib` because that is outside of Apache Software Foundation governance.

Reply:
+1 on naming it `datafusion-spark-expr` because it reflects the generic usage of the functions.

Reply:
`comet-expr`? So we will have `comet-core`, `comet-expr`, `comet-xxx`... etc.

Reply:
Another option for the crate name could be `datafusion-comet-spark-expr`.

The crate is part of the Comet project but the intention is that anyone can use it with DataFusion to emulate Spark behavior, without depending on the overall Comet project.

Reply:
I like `datafusion-comet-spark-expr` fwiw. It would be nice for all things versioned together to have the sama datafusion-comet prefix, and as the reason to use these exprs over DF’s inbuilt one is to align with spark, having that in the name would make sense to me as well.

Reply:
I have renamed the crate to `datafusion-comet-spark-expr`. 

@advancedxy @viirya @huaxingao what do you think?

Reply:
Looks okay to me.

Reply:
+1